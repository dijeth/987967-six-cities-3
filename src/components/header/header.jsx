import React from 'react';
import PropTypes from 'prop-types';
import {AppRoute} from '../../const/const.js';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import {getUserPicture, getEmail, getIsUserSuper, getAuthorizationStatus} from '../../reducers/user/selectors.js';

const Header = ({ isAuthorized, userPicture, email, isUserSuper, isActiveLogo }) => {
  const userNameBlock = isAuthorized ? 
    <span className="header__user-name user__name">{email}</span> 
    : <span className="header__login">Sign in</span>
  
  const link = isAuthorized ? AppRoute.getFavorites() : AppRoute.getLogin();

  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            {!isActiveLogo && (
              <Link to={AppRoute.getRoot()} className="header__logo-link">
                <img className="header__logo" src="/img/logo.svg" alt="6 cities logo" width="81" height="41"/>
              </Link>)}

            {isActiveLogo && (
              <a className="header__logo-link header__logo-link--active">
                <img className="header__logo" src="/img/logo.svg" alt="6 cities logo" width="81" height="41"/>
              </a>)}
          </div>
          <nav className="header__nav">
			      <ul className="header__nav-list">
			        <li className="header__nav-item user">
			          <Link to={link} className="header__nav-link header__nav-link--profile" href="#">
			            <div className={`header__avatar-wrapper ${isUserSuper ? `header__avatar-wrapper--pro` : ``} user__avatar-wrapper`}>
			              {userPicture && <img className="header__avatar user__avatar" src={userPicture} width="74" height="74" alt="User avatar" />}
			            </div>
			            {userNameBlock}
			          </Link>
			        </li>
			      </ul>
			    </nav>
        </div>
      </div>
    </header>)
};

Header.propTypes = {
  isAuthorized: PropTypes.bool.isRequired,
  userPicture: PropTypes.string,
  email: PropTypes.string,
  isUserSuper: PropTypes.bool,
  isActiveLogo: PropTypes.bool
};

const mapStateToProps = (state) => ({
  userPicture: getUserPicture(state),
  email: getEmail(state),
  isUserSuper: getIsUserSuper(state),
  isAuthorized: getAuthorizationStatus(state)
})

export default connect(mapStateToProps)(Header);
