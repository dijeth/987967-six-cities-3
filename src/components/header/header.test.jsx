import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {Header} from './header.jsx';

describe(`<Header /> snapshot test`, () => {
  it(`should be rendered correctly when the user is not authorized`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Header
            isAuthorized={false}
          />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should be rendered correctly when the user is authorized`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Header
            userPicture="img/user.jpg"
            email="aaa@ee.ee"
            isSuperUser={true}
            isAuthorized={true}
          />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should be rendered correctly when path is root`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Header
            userPicture="img/user.jpg"
            email="aaa@ee.ee"
            isSuperUser={true}
            isAuthorized={true}
            isActiveLogo={true}
            isActiveLogo={true}
          />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should be rendered correctly when path is not root`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Header
            userPicture="img/user.jpg"
            email="aaa@ee.ee"
            isSuperUser={true}
            isAuthorized={true}
            isActiveLogo={true}
            isActiveLogo={true}
          />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
