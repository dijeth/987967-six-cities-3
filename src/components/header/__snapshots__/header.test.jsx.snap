// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> snapshot test should be rendered correctly when path is not root 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper header__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="User avatar"
                  className="header__avatar user__avatar"
                  height="74"
                  src="https://htmlacademy-react-3.appspot.com/six-citiesimg/user.jpg"
                  width="74"
                />
              </div>
              <span
                className="header__user-name user__name"
              >
                aaa@ee.ee
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`<Header /> snapshot test should be rendered correctly when path is root 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper header__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="User avatar"
                  className="header__avatar user__avatar"
                  height="74"
                  src="https://htmlacademy-react-3.appspot.com/six-citiesimg/user.jpg"
                  width="74"
                />
              </div>
              <span
                className="header__user-name user__name"
              >
                aaa@ee.ee
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`<Header /> snapshot test should be rendered correctly when the user is authorized 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper header__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="User avatar"
                  className="header__avatar user__avatar"
                  height="74"
                  src="https://htmlacademy-react-3.appspot.com/six-citiesimg/user.jpg"
                  width="74"
                />
              </div>
              <span
                className="header__user-name user__name"
              >
                aaa@ee.ee
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`<Header /> snapshot test should be rendered correctly when the user is not authorized 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/login"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper  user__avatar-wrapper"
              />
              <span
                className="header__login"
              >
                Sign in
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
